crates/sol-macro/src/expand/ty.rs

```diff
@@ -98,7 +98,7 @@ pub(crate) fn rec_expand_rust_type(ty: &Type, crates: &ExternCrates, tokens: &mu
         Type::Address(span, _) => quote_spanned! {span=> #alloy_sol_types::private::Address },
         Type::Bool(span) => return Ident::new("bool", span).to_tokens(tokens),
         Type::String(span) => quote_spanned! {span=> #alloy_sol_types::private::String },
-        Type::Bytes(span) => quote_spanned! {span=> #alloy_sol_types::private::Vec<u8> },
+        Type::Bytes(span) => quote_spanned! {span=> #alloy_sol_types::private::Bytes },
```

Comment:
I think this should work, but not entirely sure @DaniPopes 

I was wondering if there's any ambiguity wrt byte[] vs bytes, we previously had this issue in abigen!

Reply:
sol_types does not have the ambiguity in usage that abigen had because it always fully specifies the solidity type via the type system. 

the goal of this PR is to remove ambiguity in the _mapping between_ rust and sol_types by establishing that `Vec<u8>` corresponds to `uint8[]` and `Bytes` corresponds to `bytes`. This allows stylus to assume the mapping is bijective and remove some user-facing complexity